lib/base_decorator.rb

```diff
@@ -0,0 +1,10 @@
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] Good job decorating your classes , however the attribute accessor should be created for the @nameable instance variable. Kindly add it 😄 
```bash
class BaseDecorator < Nameable
    attr_accessor :nameable
...
end
```